<a href="https://colab.research.google.com/github/alexDx12/gb_statistics/blob/lesson_4/lesson_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Теория вероятностей и математическая статистика

## Урок 4

## Непрерывные случайные величины. Функция распределения и функция плотности. Нормальное распределение. Центральная предельная теорема

### Домашнее задание

### Задача 1

__Постановка задачи__

Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

__Решение__

Левая (a) и правая (b) границы промежутка величины A исходя из постановки задачи:

In [ ]:
a = 200
b = 800

Среднее значение величины A исходя из свойств равномерного распределения (mean):

In [ ]:
mean = (a + b) / 2
mean

500.0

Дисперсия величины A исходя из свойств равномерного распределения (var):

In [ ]:
var = (b - a) ** 2 / 12
var

30000.0

Ответ: среднее значение величины A равно 500, её дисперсия равна 30000.

### Задача 2

__Постановка задачи__

О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.

__Решение__

Дисперсия (var) и левая граница (a) непрерывной величины B исходя из постановки задачи:

In [ ]:
var = 0.2
a = 0.5

Правая граница (b) при этом не известна исходя из постановки задачи.

Исходя из свойств непрерывной равномерно распределённой величины, её дисперсия (var) вычисляется в соотвествии со следующей формулой:

$$var=\dfrac{(b-a)^2}{12}$$

Запишем данную формулу относительно правой границы (b):
$$b=\sqrt{12 \cdot var} + a$$

Правая граница (b):

In [ ]:
b = (12 * var) ** 0.5 + a
b

2.049193338482967

Исходя из свойств непрерывной равномерно распределённой величины, её среднее (mean) вычисляется в соотвествии со следующей формулой:

$$mean=\dfrac{a + b}{2}$$

Среднее (mean):

In [ ]:
mean = (a + b) / 2
mean

1.2745966692414834

Ответ: правая граница величины B равна 2.049, её среднее равно 1.275.

### Задача 3

__Постановка задачи__

Непрерывная случайная величина X распределена нормально и задана плотностью распределения
f(x) = (1 / (4 * sqrt(2*pi))) * (exp(-(x+2)**2) / 32).
Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение).

__Решение__

Плотность распределения непрерывной случайной величины с нормальным распределением оперелеется по формуле:
$$f(x) = \dfrac{1}{\sqrt{2 \pi \sigma^2}} \cdot \exp \left(- \frac{(x - \mu)^2}{2 \sigma^2} \right)$$

Параметры $\mu$ и $\sigma$ задают, соответственно, математическое ожидание и среднее квадратическое отклонение случайной величины:
$$M(X) = \mu, \:\: D(X) = \sigma^2$$
В результате сопоставления данной формулы с формулой, заданной в постановке задачи, получены следующие результаты:
M(X) = -2
D(X) = 16
std(X) = 4.

Ответ: M(X) = -2, D(X) = 16, std(X) = 4.

### Задача 4

__Постановка задачи__

Рост взрослого населения города X имеет нормальное распределение.
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.
Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
а). больше 182 см
б). больше 190 см
в). от 166 см до 190 см
г). от 166 см до 182 см
д). от 158 см до 190 см
е). не выше 150 см или не ниже 190 см
ё). не выше 150 см или не ниже 198 см
ж). ниже 166 см.

__Решение__

Импорт библиотек и скриптов:

In [ ]:
from scipy import stats

Среднее (mean) и среднее квадратичное отклонение (std) исходя из постановки задачи:

In [ ]:
mean = 174
std = 8

Объект функций нормального распределения (norm):

In [ ]:
norm = stats.norm(mean, std)

Вопрос a:

In [ ]:
p = 1 - norm.cdf(182)
p

0.15865525393145707

Вопрос б:

In [ ]:
p = 1 - norm.cdf(190)
p

0.02275013194817921

Вопрос в:

In [ ]:
p = norm.cdf(190) - norm.cdf(166)
p

0.8185946141203637

Вопрос г:

In [ ]:
p = norm.cdf(182) - norm.cdf(166)
p

0.6826894921370859

Вопрос д:

In [ ]:
p = norm.cdf(190) - norm.cdf(158)
p

0.9544997361036416

Вопрос е:

In [ ]:
p = norm.cdf(150) + 1 - norm.cdf(190)
p

0.0241000299798092

Вопрос ё:

In [ ]:
p = norm.cdf(150) + 1 - norm.cdf(198)
p

0.002699796063260096

Вопрос ж:

In [ ]:
p = norm.cdf(166)
p

0.15865525393145707

Ответ: а) 0.15866; б) 0.02275; в) 0.81859; г) 0.68269; д) 0.95450; е) 0.02410; ё) 0.00270; ж) 0.15866.

### Задача 5

__Постановка задачи__

На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

__Решение__

Дисперсия (var) исходя из постановки задачи:

In [ ]:
var = 25

Среднее квадратичное отклонение (sigma):

In [ ]:
sigma = var ** 0.5
sigma

5.0

Относительное отклонение роста человека от математического ожидания (delta) в сигмах:

In [ ]:
delta = (190 - 178) / sigma
delta

2.4

Ответ: 2.4 сигмы.